### ISSUE TITLE
[a11y]: Unable to delete the files with Keyboard/ Mouse in Fileuploader

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16301]
### Package

@carbon/react

### Package version

v1.55.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

The file deletion functionality lacks proper accessibility support, Can not delete files using keyboard or mouse

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/docs/components-fileuploader--overview

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
